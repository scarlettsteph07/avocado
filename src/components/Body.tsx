import React from 'react'
import axios from 'axios'

import styled from '../styled'
import { API_URL } from '../lib/appConstants'
import {
  IngredientsList,
  Settings,
} from './'
import { StyledButton } from './styled/'

import { Ingredient, Payload } from '../types/'

interface Props {

}

interface State {
  isLoading: boolean,
  ingredients: Array<Ingredient>,
  payload: Payload,
}

class Body extends React.Component<Props, State> {
  state: State = {
    isLoading: false,
    ingredients: [],
    payload: {
      isCarnivore: false,
      dietPreference: 'carnivore',
      numOfOptionalIngredients: 1,
    }
  }

  componentDidMount = () => {
    this.setState({ isLoading: true })
    this.fetchIngredients(this.setPayload())
  }

  setPayload = (): Payload => {
    const { payload } = this.state
    return {
      "isCarnivore": payload.isCarnivore,
      "dietPreference": payload.dietPreference,
      "numOfOptionalIngredients": payload.numOfOptionalIngredients,
    }
  }

  fetchIngredients = (data: Object) => {
    axios.post(
      API_URL,
      data,
    )
    .then( response => {
      const { ingredients } = response.data
      this.setState({ ingredients, isLoading: false })
    })
    .catch(error => {
      console.log(error)
    })
  }

  handleOnButtonClick = () => {
    this.fetchIngredients(this.setPayload())
  }

  updateIsCarnivore = () => {
    this.setState(prevState => ({
      payload: {
        ...prevState.payload,
        isCarnivore: !this.state.payload.isCarnivore,
      }
    }), () => this.fetchIngredients(this.setPayload()))
  }

  updateNumberOfIngredients = (numOfOptionalIngredients: number) => {
    this.setState(prevState => ({
      payload: {
        ...prevState.payload,
        numOfOptionalIngredients,
      }
    }), () => this.fetchIngredients(this.setPayload()))
  }

  render() {
    const { isLoading, payload } = this.state
    return (
    <BodyStyles className="body">
      <Settings
        updateIsCarnivore={this.updateIsCarnivore}
        isCarnivore={payload.isCarnivore}
        dietPreference={payload.dietPreference}
        updateNumberOfIngredients={this.updateNumberOfIngredients}
        numOfOptionalIngredients={payload.numOfOptionalIngredients}
      />
      <IngredientsList ingredients={this.state.ingredients} />
      { !isLoading &&
        <RandomizeButton
          className="randomize-button"
          onClick={(this.handleOnButtonClick)}
        >
          Randomize!
        </RandomizeButton>
      }
    </BodyStyles>
  )}
}

export { Body }

const BodyStyles = styled.div`
  width: 100%;
  min-height: 90vh;
  box-sizing: border-box;
  font-size: 25px;
  color: ${({ theme }) => theme.textPrimary};
  background: white;
  background: linear-gradient(0, ${({theme}) => theme.snowDrift} 8%, ${({theme}) => theme.oldPaper} 8%) 0 57px;
  background-size: 100% 30px;
  display: flex;
  flex-flow: column;
  align-items: center;
  justify-content: flex-start;
  @media (min-width: ${({ theme }) => theme.medium.start} ) {
    font-size: 70px;
  }
`

const RandomizeButton = styled(StyledButton)`
  min-width: 25%;
  height: auto;
  margin: 25px auto;
  font-size: 25px;
  @media (min-width: ${({ theme }) => theme.medium.start} ) {
    font-size: 50px;
  }
`