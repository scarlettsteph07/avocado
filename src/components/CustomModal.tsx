import React, { useState } from 'react'
import Modal from 'react-modal'

import styled from '../styled'
import { StyledIconButton } from './styled'

type Props = {
  children: JSX.Element
}

export const CustomModal: React.FunctionComponent<Props> = ({
  children,
}: Props) => {
  Modal.setAppElement('#root')
  const [isModalOpen, setIsModalOpen] = useState(false)

  const openModal = (): void => {
    setIsModalOpen(true)
  }

  const closeModal = (): void => {
    setIsModalOpen(false)
  }

  const customStyles = {
    content: {
      width: '70%',
      top: '50%',
      left: '50%',
      right: 'auto',
      bottom: 'auto',
      marginRight: '-50%',
      transform: 'translate(-50%, -50%)',
      display: 'flex',
      flexFlow: 'column',
      border: '2px solid #202020',
      borderRadius: '3px',
    },
  }

  const childrenWithProps = React.Children.map(children, (child) =>
    React.cloneElement(child, { closeModal: closeModal }),
  )

  return (
    <CustomModalStyles>
      <OpenIconButton onClick={openModal} />
      <Modal
        contentLabel="Modal"
        isOpen={isModalOpen}
        onRequestClose={closeModal}
        style={customStyles}
      >
        <CloseIconButton onClick={closeModal} />
        {childrenWithProps}
      </Modal>
    </CustomModalStyles>
  )
}

const OpenIconButton = styled(StyledIconButton)`
  margin: 0;
  background-image: url(/svg/icon--plus.svg);
  align-self: flex-end;
`

const CloseIconButton = styled(StyledIconButton)`
  margin: 0;
  background-image: url(/svg/icon--x.svg);
  align-self: flex-end;
`
const CustomModalStyles = styled.div``
