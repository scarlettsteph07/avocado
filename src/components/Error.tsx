import React from 'react'
import _ from 'lodash'

import styled from '../styled'
import { StyledSlatOuter, StyledSlatInner } from './styled/'

type Props = { error: string }

export const Error: React.FunctionComponent<Props> = ({
  error,
}: Props) => {
  return (
    <ErrorStyles className="error">
      <StyledSlatOuter className="error__outer">
        <StyledSlatInner className="error__inner">
          <h1>Oops, something went wrong!</h1>
          <h3>{_.startCase(error)}</h3>
        </StyledSlatInner>
      </StyledSlatOuter>
    </ErrorStyles>
  )
}

const ErrorStyles = styled.div``
