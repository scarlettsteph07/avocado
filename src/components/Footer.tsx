import React from 'react'

import styled from '../styled'

interface Props {

}

export const Footer = () => {
  const currentTime = new Date()
  return (
    <FooterStyles className='footer'>
      &copy; {currentTime.getFullYear()} scarlettsteph07. All Rights Reserved.
    </FooterStyles>
  )
}

const FooterStyles = styled.footer`
  height: 25px;
  display: flex;
  justify-content: center;
  align-items: center;
  @media (min-width: ${({ theme }) => theme.medium.start} ) {
    height: 70px;
  }
`