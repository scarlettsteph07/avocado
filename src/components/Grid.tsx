import React from 'react'
import axios from 'axios'
import _ from 'lodash'

import styled from '../styled'
import {
  API,
  API_DEV_USER,
  APPLICATION_PATHS as PATHS,
} from '../lib/appConstants'
import { SubHeader } from './'

import { Ingredient } from '../types'

interface State {
  isLoading: boolean
  ingredients: Array<Ingredient>
}

export class Grid extends React.Component<{}, State> {
  state: State = {
    isLoading: false,
    ingredients: [],
  }
  componentDidMount = (): void => {
    this.setState({ isLoading: true })
    this.fetchAllIngredients(API_DEV_USER)
  }

  fetchAllIngredients = (userKey: string): void => {
    const axiosInstance = axios.create({
      baseURL: API.ALL.URL,
      headers: {
        'x-user-key': userKey,
      },
    })

    axiosInstance
      .get(API.ALL.URL)
      .then((response) => {
        const ingredients = response.data
        this.setState({
          ingredients: _.orderBy(
            ingredients,
            ['name', 'style'],
            ['asc'],
          ),
          isLoading: false,
        })
      })
      .catch((error) => {
        console.log(error)
      })
  }

  render(): React.ReactNode {
    const { isLoading, ingredients } = this.state
    !isLoading && console.log('ingredients: ', ingredients)
    return (
      <GridStyles className="grid">
        <SubHeader titleText="All Ingredients" />
        <div className="grid__container">
          {!isLoading &&
            ingredients.map((ingredient, index) => {
              return (
                <a
                  className="grid__item"
                  key={index}
                  href={`/${PATHS.INGREDIENTS}/${_.kebabCase(
                    ingredient.name,
                  )}`}
                >
                  <div>{ingredient.name}</div>
                </a>
              )
            })}
        </div>
      </GridStyles>
    )
  }
}

const GridStyles = styled.div`
  display: flex;
  flex-flow: column;
  .grid {
    &__container {
      display: flex;
      flex-flow: row;
      flex-wrap: wrap;
      border: 1px solid ${({ theme }) => theme.textPrimary};
    }
    &__item {
      width: 25%;
      height: 100px;
      text-align: center;
      border: 1px solid ${({ theme }) => theme.textPrimary};
      box-sizing: border-box;
      text-decoration: none;
      color: inherit;
      display: flex;
      align-items: center;
      justify-content: center;
      &:hover {
        text-decoration: underline;
        background-color: ${({ theme }) => theme.textPrimary};
        color: ${({ theme }) => theme.white};
      }
      @media (min-width: ${({ theme }) => theme.devices.tablet}) {
        font-size: 34px;
        letter-spacing: 1px;
      }
    }
  }
`
