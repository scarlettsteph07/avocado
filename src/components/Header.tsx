import React from 'react'

import styled from '../styled'
import { StyledSlatInner, StyledSlatOuter } from './styled/'
import { keyframes } from '@emotion/core'

export interface Props {
  mealTitle: string
  iconSource: string
}

export const Header: React.FunctionComponent<Props> = ({
  mealTitle,
  iconSource,
}: Props) => {
  return (
    <HeaderStyles className="header">
      <StyledSlatOuter className="header__outer">
        <StyledSlatInner className="header__inner">
          <img
            src={iconSource}
            className="header__icon"
            alt="avocado"
          />
          <h1 className="header__title">{mealTitle}</h1>
        </StyledSlatInner>
      </StyledSlatOuter>
    </HeaderStyles>
  )
}

const hithere = keyframes`
  40%, 60% { transform: rotate(-20deg); }
  50% { transform: rotate(20deg); }
  70% { transform: rotate(0deg); }
`

const HeaderStyles = styled.header`
  height: auto;
  text-align: center;
  font-size: calc(10px + 2vmin);
  color: ${({ theme }) => theme.textPrimary};
  .header {
    &__inner {
      display: flex;
      justify-content: space-between;
      align-items: center;
      @media (min-width: ${({ theme }) => theme.devices.tablet} ) {
        flex-direction: row;
        justify-content: space-between;
      }
    }
    &__icon {
      height: 50px;
      pointer-events: none;
      /* animation: ${hithere} 1s ease infinite; */
      @media (min-width: ${({ theme }) => theme.devices.tablet} ) {
        height: 100px;
      }
      @media (min-width: ${({ theme }) => theme.devices.desktop} ) {
        height: 160px;
      }
    }
    &__title {
      letter-spacing: 0.4em;
      margin: 0px !important;
      text-transform: uppercase;
      @media (min-width: ${({ theme }) => theme.devices.tablet} ) {
        letter-spacing: 0.2em;
        font-size: 65px;
      }
      @media (min-width: ${({ theme }) => theme.devices.desktop} ) {
        font-size: 100px;
      }
    }
  }
`
