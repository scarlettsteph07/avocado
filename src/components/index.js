export * from './Body'
export * from './Footer'
export * from './Header'
export * from './IngredientsGrid'
export * from './IngredientsList'
export * from './IngredientPage'
export * from './Picker'
export * from './Recipe'
export * from './RecipeContainer'
export * from './Settings'
export * from './SubHeader'
