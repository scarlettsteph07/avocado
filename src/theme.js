export let theme = {
  small: {
    outer: '320px',
    inner: '320px',
    start: '0px',
    end: '640px',
  },
  medium: {
    inner: '640px',
    start: '641px',
    end: '1024px'
  },
  large: {
    before: '1139px',
    inner: '1140px',
    start: '1141px',
    end: '1440px',
  },
  xlarge: {
    outer: '1440px',
    inner: '1140px',
    start: '1441px'
  },
  devices: {
    desktop: '926px',
    tablet: '768px',
  },

  baseFont: '"Scarlett", "Annie Use Your Telescope", cursive',
  secondaryFont: '"Helvetica Neue", Helvetica, sans-serif',

  textPrimary:              '#202020',
  transparent:              'transparent',
  transparentOverlay:       'rgba(0,0,0,0.2)',
  semiTransparentOverlay:   'rgba(0,0,0,0.05)',
  black:                    '#000000',
  white:                    '#ffffff',
  emerald:                  '#4ed164',
  lightGray:                '#D9DADC',
  red:                      '#FF0000',
}
