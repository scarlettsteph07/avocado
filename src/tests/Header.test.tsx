import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'emotion-theming'

import { Header } from '../components/'
import { theme } from '../theme'

describe('Header component', () => {
  test('should match snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Header
          mealTitle="toast &amp; avocado"
          iconSource="/img/avocado.png"
        />
      </ThemeProvider>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
