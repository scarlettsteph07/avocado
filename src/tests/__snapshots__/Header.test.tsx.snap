// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<header
  className="header css-m90vym"
>
  <section
    className="header__outer css-gvh4h9"
  >
    <div
      className="header__inner css-12rf414"
    >
      <img
        alt="avocado"
        className="header__icon"
        src="/img/avocado.png"
      />
      <h1
        className="header__title"
      >
        toast & avocado
      </h1>
    </div>
  </section>
</header>
`;
